src/kernels/gpu_reference_kernel/naive_conv.cpp

```diff
@@ -234,21 +236,27 @@ inline __device__ void naive_conv_fwd_nchw(const src_data_t* __restrict__ p_in,
         {
             size_t o_idx = static_cast<size_t>(iho) * wo + static_cast<size_t>(iwo);
 
-            p_out[o_idx] = cast_to<acc_data_t, dst_data_t>(value);
+            acc_data_t val_alpha_beta =
```

Comment:
This is good but I don't think we are quite complying with the spec. I was wrong to suggest that we can use `acc_data_t` for `alpha/beta`. In reality, it should be `float` or `double`. I think the more correct solution would be to deduce the type of `alpha/beta` from `src_data_t`. This is where we use your type-deduction logic :).